src/renderer/src/routes/Onboarding/DeviceNamingScreen.tsx

```diff
@@ -51,110 +54,62 @@ export const Route = createFileRoute('/Onboarding/DeviceNamingScreen')({
 	component: DeviceNamingScreenComponent,
 })
 
-const Container = styled('div')({
-	display: 'flex',
-	flexDirection: 'column',
-	alignItems: 'center',
-	justifyContent: 'center',
-	height: '100%',
-	backgroundColor: DARK_COMAPEO_BLUE,
-})
-const ContentBox = styled('div')({
-	backgroundColor: 'rgba(255, 255, 255, 0.94)',
-	border: `1px solid ${BLUE_GREY}`,
-	borderRadius: 8,
-	padding: 20,
-	width: '55%',
-	textAlign: 'center',
-	boxShadow: '0px 4px 4px 0px rgba(0, 0, 0, 0.02)',
-})
-const BodyTextWrapper = styled('div')({
-	maxWidth: '45%',
-	margin: '16px auto 0',
-	textAlign: 'center',
-})
-const ButtonContainer = styled('div')({
-	display: 'flex',
-	justifyContent: 'center',
-	gap: 15,
-	marginTop: 63,
-	padding: '0 20px',
-})
 const StyledImage = styled('img')({
-	marginBottom: 20,
 	width: 60,
 	height: 48,
 })
+
 const InputWrapper = styled('div')({
-	marginTop: 24,
-	marginBottom: 160,
+	marginTop: 12,
+	marginBottom: 12,
 	display: 'flex',
 	flexDirection: 'column',
 	alignItems: 'center',
+	flexGrow: 1,
 })
+
 const StyledTextField = styled(TextField)({
 	width: '100%',
-	maxWidth: '400px',
+	maxWidth: 400,
 	backgroundColor: WHITE,
-	marginTop: 20,
+	marginTop: 12,
 })
+
 const CharacterCount = styled(Text, {
 	shouldForwardProp: (prop) => prop !== 'error',
-})<{ error: boolean }>`
-	margin-top: 8px;
-	color: ${({ error }) => (error ? RED : BLACK)};
-	width: 100%;
-	max-width: 400px;
-	text-align: right;
-`
+})<{ error: boolean }>(({ error }) => ({
+	marginTop: 12,
+	color: error ? RED : BLACK,
+	width: '100%',
+	maxWidth: 400,
+	textAlign: 'right',
+}))
 
 export function DeviceNamingScreenComponent() {
 	const navigate = useNavigate()
 	const { formatMessage } = useIntl()
 	const [deviceName, setDeviceName] = useState('')
-	const [error, setError] = useState(false)
+	const [inputError, setInputError] = useState(false)
 	const [errorMessage, setErrorMessage] = useState('')
 	const setDeviceNameMutation = useEditDeviceInfo()
 
-	function countGraphemes(text: string): number {
-		const segmenter = new Intl.Segmenter(undefined, {
-			granularity: 'grapheme',
-		})
-		let result = 0
-		for (const _ of segmenter.segment(text)) result++
-		return result
-	}
-
-	function getUtf8ByteLength(text: string): number {
-		return new TextEncoder().encode(text).length
-	}
-
 	const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
 		const value = event.target.value
-		const graphemeCount = countGraphemes(value.trim())
-		const byteLength = getUtf8ByteLength(value.trim())
-		let error = false
-
-		if (
-			graphemeCount > DEVICE_NAME_MAX_LENGTH_GRAPHEMES ||
-			byteLength > DEVICE_NAME_MAX_BYTES
-		) {
-			error = true
-		} else {
-			if (value.trim().length === 0) {
-				error = true
-			}
-			setDeviceName(value)
+		const localError: boolean = checkForError(
+			value.trim(),
+			DEVICE_NAME_MAX_LENGTH_GRAPHEMES,
+		)
+		setDeviceName(value)
+		if (localError !== inputError) {
+			setInputError(localError)
```

Comment:
if there is an error, do we want to set the name? Or should we just set the device name when no error

Reply:
I am doing that because when there is a blank there (someone erases all of the letters), there is an error **and** we need to set the device name to a blank. If I don't set the device name in that case a user can never erase that last letter.